dbt/include/redshift/macros/materializations/incremental_merge.sql

```diff
@@ -65,3 +65,50 @@
     )
 
 {% endmacro %}
+
+{% macro redshift__get_incremental_microbatch_sql(arg_dict) %}
+    {#-
+        Technically this function could just call out to the default implementation of delete_insert.
+        However, the default implementation requires a unique_id, which we actually do not want or
+        need. Thus we re-implement delete insert here without the unique_id requirement
+    -#}
+
+    {%- set target = arg_dict["target_relation"] -%}
+    {%- set source = arg_dict["temp_relation"] -%}
+    {%- set dest_columns = arg_dict["dest_columns"] -%}
+    {%- set predicates = [] -%}
+
+    {%- set incremental_predicates = [] if arg_dict.get('incremental_predicates') is none else arg_dict.get('incremental_predicates') -%}
+    {%- for pred in incremental_predicates -%}
+        {% if "DBT_INTERNAL_DEST." in pred %}
+            {%- set pred =  pred | replace("DBT_INTERNAL_DEST.", target ~ "." ) -%}
+        {% endif %}
+        {% if "dbt_internal_dest." in pred %}
+            {%- set pred =  pred | replace("dbt_internal_dest.", target ~ "." ) -%}
+        {% endif %}
+        {% do predicates.append(pred) %}
+    {% endfor %}
+
+    {% if not model.config.get("__dbt_internal_microbatch_event_time_start") or not model.config.__dbt_internal_microbatch_event_time_end -%}
```

Comment:
nit: not using model.config.get for the end time. let's be consistent across the two accesses

Reply:
Good catch!

I was curious how this happened and did a little digging. In [f7899e0](https://github.com/dbt-labs/dbt-redshift/pull/924/commits/f7899e0b82b59200abb01524c44487b40fb56731) we [removed the old separate if statements](https://github.com/dbt-labs/dbt-redshift/pull/924/commits/f7899e0b82b59200abb01524c44487b40fb56731#diff-b2feb41ebf7b20e5cdaa6c29172500c8176d643a8a86025e1f58111e7bb58530L93) that added the predicates and added this new combined if statement. The old if statements came from the [original snowflake implementation](https://github.com/dbt-labs/dbt-snowflake/pull/1179/files#diff-03ee8d11934e66a23fa4ab1a018378a9533816c757ed0c709fad89ada1d12d79R60). I think did it that way originally because our [initial implementation of batches in core](https://github.com/dbt-labs/dbt-core/pull/10677) the [start time was not guaranteed](https://github.com/dbt-labs/dbt-core/pull/10677/files#diff-99215b70f7814227b4602a6cdec5a2e33d02023f16e258dd6ae1808bf818bbc3R69). And so the `get` was probably to handle when the start time was `None` (which now never happens, because the start time is requried for every batch)